src/main/resources/plugin.properties

```diff
@@ -1,3 +1,3 @@
 name=dependency-track-companion-plugin
 vendor=Liftric
-version=1.0.0
+version=development
```

Comment:
Why do you still need this property?

Reply:
Ah, don't generate code/data which gets added to git. Better:

```
Subject: [PATCH] x
---
Index: src/main/resources/plugin.properties
===================================================================
diff --git a/src/main/resources/plugin.properties b/src/main/resources/plugin.properties
deleted file mode 100644
--- a/src/main/resources/plugin.properties	(revision 78f58847405c44c98717bc5300b22b8c3f89f3ba)
+++ /dev/null	(revision 78f58847405c44c98717bc5300b22b8c3f89f3ba)
@@ -1,3 +0,0 @@
-name=dependency-track-companion-plugin
-vendor=Liftric
-version=development
Index: build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle.kts b/build.gradle.kts
--- a/build.gradle.kts	(revision 78f58847405c44c98717bc5300b22b8c3f89f3ba)
+++ b/build.gradle.kts	(date 1689313462395)
@@ -8,6 +8,11 @@
     alias(libs.plugins.versioning)
 }
 
+group = "com.liftric"
+version = with(versioning.info) {
+    if (branch == "HEAD" && dirty.not()) tag else full
+}
+
 repositories {
     mavenCentral()
     gradlePluginPortal()
@@ -23,12 +28,8 @@
     buildBeforeUp.set(true)
 }
 
-group = "com.liftric"
-version = with(versioning.info) {
-    if (branch == "HEAD" && dirty.not()) tag else full
-}
-
 val integrationTest = sourceSets.create("integrationTest")
+
 tasks {
     val test by existing
     withType<Test> {
@@ -49,26 +50,31 @@
     }
     dockerCompose.isRequiredBy(integrationTestTask)
 
+    val pluginPropertiesBuildFolder  = file("$buildDir/compileProperties/")
     val propertiesTask = register<WriteProperties>("writePluginProperties") {
-        outputFile = file("src/main/resources/plugin.properties")
+        outputFile = pluginPropertiesBuildFolder.resolve("plugin.properties")
         property("vendor", "Liftric")
         property("name", rootProject.name)
         property("version", rootProject.version)
     }
+    sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME).resources.srcDir(pluginPropertiesBuildFolder)
+
+    withType(ProcessResources::class.java) {
+        dependsOn(propertiesTask)
+    }
 }
 
 tasks.named("publishPlugins") {
     dependsOn("writePluginProperties")
 }
 
-
 gradlePlugin {
     testSourceSets(integrationTest)
     plugins {
         create("dependency-track-companion-plugin") {
             id = "$group.$name"
-            implementationClass = "${rootProject.group}.dtcp.DepTrackCompanionPlugin"
-            displayName = rootProject.name
+            implementationClass = "$group.dtcp.DepTrackCompanionPlugin"
+            displayName = name
             description = "Common tasks for Dependency Track interaction, like SBOM upload or VEX Generation"
         }
     }
```
Now the file is always in sync with the project version :+1:

Reply:
Nice Solution! Tested it locally, everything fine.